src/entity.ts

```diff
@@ -134,7 +134,7 @@ export class Entity {
 		if (!this._context)
 			this._context = context;
 		// Ensure provided context waits on the existing context to be ready
-		else
+		else if (this._context !== context)
```

Comment:
There was a sneaky bug where revisiting an entity would result in a deadlock (the context would be waiting on itself to be ready).

Reply:
Makes sense